### PR TITLE
Update addresses to EIP-55 format

### PR DESCRIPTION
**Pull request by theosirian**
- Updates addresses to follow EIP-55 format
- Add EIP-55 negative parsing case

### CODE REVIEW

**Review by sbihel — COMMENTED** [review]
Could we keep the formatting, the indentation and not try to align columns, so in the future the addition of a single test only adds one line, making the git history clean?

**Review by sbihel — CHANGES_REQUESTED** [review]

File: test/parsing_negative.json [hunk 1]
```diff
@@ -1,7 +1,8 @@
...
+    "out of order fields": "service.org wants you to sign in with your Ethereum account:\n0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2\n\nI accept the ServiceOrg Terms of Service: https://service.org/tos\n\nVersion: 1\nNonce: 32891757\nURI: https://service.org/login\nIssued At: 2021-09-30T16:25:24.000Z",
+    "wrong version": "service.org wants you to sign in with your Ethereum account:\n0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2\n\nI accept the ServiceOrg Terms of Service: https://service.org/tos\n\nURI: https://service.org/login\nVersion: 2\nNonce: 32891757\nIssued At: 2021-09-30T16:25:24.000Z",
+    "address not eip-55": "service.org wants you to sign in with your Ethereum account:\n0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2\n\nI accept the ServiceOrg Terms of Service: https://service.org/tos\n\nURI: https://service.org/login\nVersion: 2\nNonce: 32891757\nIssued At: 2021-09-30T16:25:24.000Z"
```
[the remark this answers is not in the document]
Comment (awoie) [reply test/parsing_negative.json:7]: Version has to be 1 according to spec (for now and for the defined abnf)

[the remark this answers is not in the document]
Comment (awoie) [reply test/parsing_negative.json:7]: an example of non-compliant address is the followiing: `0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2`

**Comment by w4ll3** [comment]
adressed at #62

**Pull request closed without merging**